### ISSUE TITLE
"video_mode_ntsc" and "video_mode_pal" ignored when "vsync_adjust=0"

### ISSUE THREAD

**Issue by jlancaster86** [MiSTer-devel/Main_MiSTer#506]
Sorry if this is a stupid question, but is there a reason the specific NTSC and PAL modes aren't honoured when "vsync_adjust" is set to "0"? Can the mode not be changed when a core is switched between NTSC and PAL?

**Comment by sorgelig** [comment]
this is by design because there is no point to switch output mode when it's not synced to original video. 
From my observation it's even better to use NTSC mode on HDMI when core outputs PAL - vide is less jerky. When both video are PAL or NTSC then every several seconds you may notice big jerk.

**Comment by paulb-nl** [comment]
50 to 60Hz must be much worse than 50 to 50. Do you mean constantly skipping frames looks better to you than skipping a frame every few seconds?

**Comment by sorgelig** [comment]
yes. you can try 240p Suite (some console cores have) and use scenery scroll test. 50 on 60 looks surprisingly more smooth. It jerks fast and not so annoying. 50 on 50 (or 60 on 50) jerks every several seconds and disturbs a lot. That's because refresh is very close but doesn't match.

**Comment by jlancaster86** [comment]
Does that scrolling behaviour vary between cores? I've just done some more testing in the NES core, setting it to PAL:

Using the Grid Scroll Test in 240p Test Suite, the scrolling is smooth when I set "video_mode" to "7" (1280x720@**50**), but jerky when left on "0" (1280x720@**60**). When playing a regular game (I'm testing with _Astérix_), a 50 Hz video mode definitely looks better on my displays.

**Comment by sorgelig** [comment]
then why don't use vsync_adjust=1?

**Comment by jlancaster86** [comment]
Because my Samsung TV was exhibiting screen tearing/glitching with "vsync_adjust=1", but not "vsync_adjust=0". But after a lot more fiddling, it seems that only applies to HDMI inputs 1–3 (along with some colour/dimming issues), and I can go back to "vsync_adjust=1" if I use HDMI input 4 instead (the special "HDMI 2.0" port, and apparently the only one that doesn't screw everything up).

I guess I no longer have a horse in this race, but it still seems like unintuitive behaviour.

**Comment by sorgelig** [comment]
i will see if it will be reliable because vsync_adjust=0 basically means "set resolution and don't bother TV anymore". Many TVs and most monitors don't support 50hz mode.

**Comment by jlancaster86** [comment]
Don't all modern TVs support 50 Hz? They seem to here in PAL Land (I'm in Australia), and I'd expect the same to be true in the SECAM world, and that's most of the world right there. DVD players still use 50 Hz modes after all, and I'm pretty sure DVB-T broadcasts do too.

Yeah, PC monitors traditionally don't, but VRR monitors are on the rise. I have a G-SYNC monitor that seems to handle whatever I throw at it.

**Comment by birdybro** [comment]
Yes, modern TV's almost universally support 50hz for all sorts of legitimate reasons. Your TV though seems to just have issues depending on which port is used, which is actually very common the last few years as they are applying different HDMI specs to different ports. It's kind of annoying actually. You can find all sorts of issues people have with ARC causing conflicts on the ARC port, then HDMI 2.0 not working on others, HDMI 1.1 not working on others, etc... It's a mess.

I'm glad you figured out what the solution was though, to use HDMI-4, and thank you for posting it here so it can be referenced later for other Samsung TV users. I can confirm, Samsung tv's have very bad compatibility with less-standard refresh rates for vsync_adjust=1 and vsync_adjust=2. This is an inherent flaw, I've confirmed it with a Retrotink 5x Pro with original hardware, had the same jittery sync problems with a Sega Genesis when I enabled the RT5x-pro's equivalent to vsync_adjust=1.

This can probably be closed now @jlancaster86 :)

**Comment by jlancaster86** [comment]
I don't want to clog up this issue with tails of my Samsung TV, so I'll keep this brief. I did get everything working properly again on the original HDMI input (somehow), only for it to suddenly break again today. I can't get it working properly on HDMI-4 either. The Game Mode settings are apparently just buggy on these TVs and sometimes things work, and sometimes they don't. So, I'm back to "vsync_adjust=0" for the sake of my sanity.

In any case, the current behaviour of "vsync_adjust=0" is wrong; it should be honouring "video_mode_ntsc" and "video_mode_pal" when they're set. I'll just have to create an alternate INI file with "video_mode" set to a 50 Hz mode if there's no interest in changing this behaviour.

**Comment by c0d3h4x0r** [comment]
I would also like to see auto-switching to PAL/50Hz mode when `vsync_adjust=0`.  Most modern TVs support 50Hz mode (all three of mine do), but some require *exact* 50Hz (one of my TVs will fail to achieve picture sync if you feed it something like 49.7Hz or 50.2Hz, so I can't use `vsync_adjust=1` with it).  50Hz content matches best to a 50Hz display mode (zero judder).

**Comment by birdybro** [comment]
This can probably be closed. It's a wontfix since vsync_adjust=0 is intended this way.

**Comment by MarkTreeSMS** [comment]
To add as I see this is reopened, I was looking for a way to make "vsync_adjust=0" target 50Hz instead of 60Hz, which I hope falls under this issue.
My situation is attempting to send PAL content from MiSTer to a wireless transmitter. The hardware struggles to meet the more stringent vsync_adjust settings, but as I need it to be PAL and only want to send PAL games across, a similar framebuffer mode would remove the occasional drops I have at the moment and be ideal. Avoid auto-switching would hopefully help against some of the concerns listed above too. I'd just like to set it to a fixed value, but for 50Hz content. Thanks

**Comment by sorgelig** [comment]
sorry for reopening. Probably it was a github glitch.
